activesupport/lib/active_support/cache.rb

```diff
@@ -459,7 +459,17 @@ def fetch(name, options = nil, &block)
             instrument(:read, name, options) do |payload|
               cached_entry = read_entry(key, **options, event: payload)
               entry = handle_expired_entry(cached_entry, key, options)
-              entry = nil if entry && entry.mismatched?(normalize_version(name, options))
+              if entry
+                if entry.mismatched?(normalize_version(name, options))
+                  entry = nil
+                else
+                  begin
+                    entry.value
+                  rescue DeserializationError
+                    entry = nil
+                  end
+                end
+              end
```

Comment:
```suggestion
                if entry && entry.mismatched?(normalize_version(name, options))
                  entry = nil
                else
                  begin
                    entry.value
                  rescue DeserializationError
                    entry = nil
                  end
                end
```
@casperisfine Do we really the nested if condition?

Reply:
Yes, in you case you don't handle `entry` being `nil`.